File: p1507.py
class Solution:
    def reformatDate(self, date: str) -> str:
       lessThan10 = len(date) == 12
       #print(len(date))
       offset = 0 if lessThan10 else 1
       d = int(date[0:1+offset])
       mm = {}
       for i, v in enumerate(["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]):
           mm[v] = i+1
       #print(date[4+offset:7+offset])
       m = mm[date[4+offset:7+offset]]
       y = int(date[-4:])
       return "{}-{:02}-{:02}".format(y, m, d)

date = "20th Oct 2052"
s = Solution()
print(s.reformatDate(date))
date = "6th Jun 1933"
print(s.reformatDate(date))
date = "26th May 1960"
print(s.reformatDate(date))


       
# 
class Solution:
    def reformatDate(self, date: str) -> str:
        # create a dict
        # handle 01 here
        Month = {"Jan":"01", "Feb":"02", "Mar":"03", "Apr":"04", "May":"05", "Jun":"06", "Jul":"07", "Aug":"08", "Sep":"09", "Oct":"10", "Nov":"11", "Dec":"12"}
        date_ = date.split(" ")
        # date_[2] is year
        # date_[1] is month
        # date_[0] is day
        # 1st or 12nd
        # len date_[0] == 3 or 4
        return '-'.join([date_[2], Month[date_[1]], date_[0][:-2] if len(date_[0][:-2]) == 2 else '0' + date_[0][:-2]])

File: p1507.cc
#include "common.h"
class Solution {
public:
  string reformatDate(string date) {
      vector<string> month = {"Jan", "Feb", "Mar", "Apr", "May", "Jun",
                              "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
      unordered_map<string, int>  map_month;
      for (int i = 0; i < 12; ++i) {
           map_month[month[i]] = i;
      }
      //Day Month Year
      int offset = 0; // one digit in
      if (isdigit(date[1])) {
          offset = 1;
      }

      int p = 3 + offset;
      int d = stoi(date.substr(0, p));
      p++;
      int m = map_month[date.substr(p, 3)] + 1;
      p += 3;
      p ++;
      int y = stoi(date.substr(p));
      string result = to_string(y) + (m < 10 ? "-0" : "-") + to_string(m) + (d < 10 ? "-0" : "-") + to_string(d);
      return result;
  }
};


int main() {

    string date = "20th Oct 2052";
    date = "6th Jun 1933";
    date = "26th May 1960";
	Solution s;
    cout << s.reformatDate(date) << endl;
    return 0;
}
